ignore/ignore.go

```diff
@@ -0,0 +1,73 @@
+// package ignore provides helpers to support ignore-files similar to .gitignore
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+var (
+	LineComment = "#"
+	IgnoreFile  = ".sourcegraph/sourcegraphignore"
+)
+
+type IgnoreMatcher struct {
+	IgnoreList []string
+	Strip      int
+}
+
+// StripComponents removes the specified number of leading path
+// elements. Pathnames with fewer elements will return the empty string.
+func StripComponents(path string, count int) string {
```

Comment:
this seems unused now that we don't do archive. Even if we do end up wanting it, should it be exported?

Reply:
Let me check if Searcher requires it. If not I will remove it.